src/llm/apis/openai_completions.cpp

```diff
@@ -75,32 +76,147 @@ absl::Status OpenAIChatCompletionsHandler::parseCompletionsPart() {
+                            size_t offset = pos + pattern.length();
+                            if (pos == std::string::npos) {
+                                offset = 0;
+                            }
+                            std::string decoded;
+                            if (!absl::Base64Unescape(std::string_view(url.data() + offset, url.size() - offset), &decoded)) {
+                                return absl::InvalidArgumentError("Invalid base64 string in request");
+                            }
+                            cv::Mat rawData(1, decoded.size(), CV_8UC1, (void*)decoded.data());
+                            cv::Mat image;
+                            try {
+                                image = cv::imdecode(rawData, cv::IMREAD_UNCHANGED);
+                            } catch (const cv::Exception& e) {
+                                return absl::InvalidArgumentError("Error during string to mat conversion");
+                            }
+                            std::vector<size_t> shape;
+                            shape.push_back(image.rows);
+                            shape.push_back(image.cols);
+                            shape.push_back(image.channels());
+                            auto type = getOvTypeFromMatType(image.depth());
+                            if (type == ov::element::undefined) {
+                                return absl::InvalidArgumentError("Image type is invalid");
+                            }
+                            ov::Tensor tensor(type, shape);
+                            if (image.total() * image.elemSize() != tensor.get_size()) {
+                                return absl::InvalidArgumentError("Image size invalid");
+                            }
+                            memcpy((char*)tensor.data(), (char*)image.data, image.total() * image.elemSize());
+                            request.images.push_back(tensor);
+                        } else {
+                            return absl::InvalidArgumentError("Unsupported content type");
+                        }
+                    }
+                    if (contentText.IsNull()) {
+                        return absl::InvalidArgumentError("Invalid message structure - content array should contain \"text\" object");
+                    }
+                    member->value = contentText;
+                } else {
+                    return absl::InvalidArgumentError("Invalid message structure - content should be string or array");
+                }
+            }
         }
     }
+    if (jsonChanged) {
+        StringBuffer buffer;
+        Writer<StringBuffer> writer(buffer);
+        doc.Accept(writer);
+        request.processedJson = buffer.GetString();
+    }
+    return absl::OkStatus();
+}
+
+const std::string& OpenAIChatCompletionsHandler::getProcessedJson() const {
+    return request.processedJson;
+}
+std::vector<ov::Tensor> OpenAIChatCompletionsHandler::getImages() const {
```

Comment:
can we return const reference to the vector? I know ov::Tensor is not heavy, but constructing the vector is